### DISCUSSION TITLE
Does it need to put both contextHolder in jsx?

### DISCUSSION THREAD

**Discussion by evgeniyworkbel** [ant-design/ant-design#42816] [Q&A] [answered]
When component use Message and Notification antd components together, does it need to put both contextHolder in my component jsx?

I mean situations where Message and Notification is created through hooks (useMessage, useNotification)

**Comment by BoyYangzai** [comment, marked as answer]
@evgeniyworkbel 
Hello bro
I think It's is truly need both, try this demo
https://codesandbox.io/s/hooks-diao-yong-tui-jian-antd-5-5-2-forked-uovru7?file=/demo.tsx
if you don't use the `contextNotification`, this `notification instance` will not show
Only the `message instance` will show successfully

**Reply by evgeniyworkbel** [reply]
many thanks! I got it
